rott/modexlib.c

```diff
@@ -209,9 +211,37 @@ void TurnOffTextCursor ( void )
 =
 ====================
 */
+
+static Uint64 next_time = 0;
+
+static void time_start(void)
+{
+	next_time = SDL_GetTicks() + VBLCOUNTER;
+}
+
+static Uint64 time_left(void)
+{
+	Uint64 now = SDL_GetTicks();
+
+	if (next_time <= now)
+		return 0;
+	else
+		return next_time - now;
+}
+
 void WaitVBL( void )
 {
-	SDL_Delay (16667/1000);
+	// was:
+	// SDL_Delay (16667/1000);
+
+	if (next_time == 0)
+	{
+		time_start();
```

Comment:
So, this means, the first time this function is called, it doesn't wait at all?

Reply:
Correct. I didn't think that'd be a huge deal, since it's called multiple times during a normal frame.

Reply:
But it still feels wrong for a waiter function not to wait.